### PR TITLE
[ENH] remove deprecated `scipy` Kulsinski distance from tests

### LINKED ISSUE #10190: [BUG] scipy not including the `kulsinski` or `kulczynski1`
**Describe the bug**
in file `test_scipy_dist.py` has a helper fun `get_kul_name()` it imports the `kulczynski1`. 
if the import failes then it goes to `kulsinski` but they both are not there in the new version of `scipy` that i `1.17`

due to this there would be import and value errors raised as found when i ran the tests in the main
Image

**To Reproduce**
run the scipy test with scipy 1.17.1 installed:
```python
python -m pytest sktime/dists_kernels/tests/test_scipy_dist.py::test_scipydist
```

**Expected behavior**
test should pass under scipy 1.17 by excluding removed metrics such as `kulsinski`, `kulczynski1`,

**Versions**
<details>
System:
    python: 3.12.3 (main, Mar 23 2026, 19:04:32) [GCC 13.3.0]
executable: /mnt/f/sktime/venv/bin/python
   machine: Linux-6.6.87.2-microsoft-standard-WSL2-x86_64-with-glibc2.39

Python dependencies:
          pip: 24.0
       sktime: 0.40.1
      sklearn: 1.7.2
       skbase: 0.13.1
        numpy: 2.3.5
        scipy: 1.17.1
       pandas: 2.3.3
   matplotlib: None
       joblib: 1.5.3
huggingface-hub: None
        numba: None
     pmdarima: None
pytorch-forecasting: None
   skforecast: None
        skpro: 2.12.0
statsforecast: None
  statsmodels: None
 transformers: None
      tsfresh: None
      tslearn: None
        torch: None
   tensorflow: None
(venv) direk@dkakkar:/mnt/f/sktime$ 
</details>

### PR DESCRIPTION
**Pull request by direkkakkar319-ops**
#### Reference Issues/PRs
Fixes #10190

#### What does this implement/fix? Explain your changes.
the scipy new version did not support the importing of `kulsinski` and `kulczynski1` due to this the wrong import happend that lead to failing test and raise valueError.

to fix this i made a fall back function `_scipy_available()` that we can use in `_get_Kul()` and test whether the metric name is there or not.

then added `_METRIC_CANDIDATES` that has filter function that maps the input to `None` if the metric name is not there in the `get_kul()` function
therefore `METRIC_VALUES` uses it and only the metrics available will be used 

this will lead to the test case to pass

tests pass locally
Screenshot 2026-05-06 103456

### CODE REVIEW

**Comment by fkiraly** [comment]
Will replace by https://github.com/sktime/sktime/pull/10252, but am merging this for credit.

**Comment by direkkakkar319-ops** [comment]
Thank you for the merge :)
